spec/controllers/reactivate_account_controller_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+describe ReactivateAccountController do
+  describe 'before_actions' do
+    it 'requires the user to be logged in' do
+      expect(subject).to have_actions(
+        :confirm_two_factor_authenticated
+      )
+    end
+  end
+
+  describe '#index' do
+    before do
+      allow(subject).to receive(:confirm_two_factor_authenticated).and_return true
```

Comment:
Ok so we should definitely try to use as much "real" code as possible and minimize stubbing of specific filters and methods like this.

I would remove these 3 stubs and do:

```ruby
let(:user) { build(:user, profiles: profiles) }
let(:profiles) { [] }
before { stub_sign_in(user) }
```

And ditto for the spec below.

Then we can override the `let(:profiles)` with an array of profile objects to match the various scenarios we want to test. (it might have to be a `create(:user ...` to make sure they get written to the DB, but same idea